random-menu-generator.rb

```diff
@@ -0,0 +1,784 @@
+    "Build a Pinky and the Brain random facts app.",
+
+    "Reprogram a roomba.",
+    "Make a clone of angry birds!",
+    "Write an app based around the concept of abstract art and the word Pinkasso.",
+    "Write a voice-to-speech interpreter, which will take in the words said and output the words in the typical voice and syntax of Morgan Freeman."
+  ]
+
+  activities = [recreation, programming_projects, world_domination]
+
+
+  if which_ideas == "hungry"
+    return food
+
+  elsif which_ideas == "bored"
+    return activities
+
+  elsif which_ideas == "dishes"
+    return dishes
+  end
+end
+
+#---------------------- END REDONKULOUS DATABASE THING -------------------------
+#-------------- BEGIN DATABASE RETRIEVE, PICK, & DISPLAY THINGS ----------------
+
+# takes dataset in array form and a number of ideas wanted from that dataset.
+# checks if the dataset is designed for display with individual or multiple
+# items on each line. checks for & removes repeat items, then assigns a maximum
+# number of items user can request from the dataset. loops through picking
+# random items from the dataset until the desired number of items are chosen.
+# finally, returns the random items to the requesting function call.
+def select_random_ideas(array_of_ideas, how_many_ideas_desired)
+  $user_wants_too_many = false
+  random_selection_of_ideas = []
+  max_value = 0
+
+  # MAX VALUE individual
+  if $individual
+    # make UNIQUE
+    array_of_ideas.flatten!.uniq!
+    # final MAX VALUE
+    max_value += array_of_ideas.length
+
+  # MAX VALUE combination
+  else
+    # make UNIQUE
+    count = 0
+    3.times do
+      array_of_ideas[count].uniq!
+      count += 1
+    end
+
+    # final MAX VALUE
+    max_value += assign_max_value(array_of_ideas)
+  end
+
+  # adjusts number wanted items if user wants more than exist
+  if how_many_ideas_desired > max_value
+    how_many_ideas_desired = max_value
+    $user_wants_too_many = true
+  end
+
+
+  # RANDOM individual
+  if $individual
+    until random_selection_of_ideas.length == how_many_ideas_desired
+      which_random_index = (rand * array_of_ideas.length).floor
+      which_random_idea = array_of_ideas.slice!(which_random_index)
+      random_selection_of_ideas.push(which_random_idea)
+    end
+
+  # RANDOM combination
+  else
+    # initialize loop count to zero
+    count = 0
+    3.times do
+      # make space for the items to go into:
+      random_selection_of_ideas.push([])
+
+      # these two for readability of following block:
+      from_array = array_of_ideas[count]
+      to_array = random_selection_of_ideas[count]
+
+      # do actual random selection & move
+      until to_array.length == how_many_ideas_desired
+        which_random_index = (rand * from_array.length).floor
+        which_random_idea = from_array.slice!(which_random_index)
+        to_array.push(which_random_idea)
+      end
+
+      # increment loop count
+      count += 1
+    end
+  end
```

Comment:
As to your question about whether you should do more to prevent duplicates in combinations, if you compile the final string in this block rather than keeping the values in separate nested arrays, you could more easily check whether you have duplicates in the string you're outputting and grab a different random value, if needed.